scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala

```diff
@@ -370,11 +374,48 @@ case class Args(
       } else {
         classpath
       }
-    Classpath(targetrootClasspath) ++ baseClasspath
+    val finalClassPath = Classpath(targetrootClasspath) ++ baseClasspath
+    val badEntries = finalClassPath.entries.filterNot {
+      ensureValidClasspathEntry
+    }
+    if (badEntries.isEmpty) {
+
+      Configured.ok(finalClassPath)
+    } else {
+      val extensions = validClasspathFileExtension.mkString("[", ",", "]")
+      val prefix = s"""|Invalid classpath entry. Make sure your classpath
+        |contains only directories or file that ends with
+        |the following extension: ${extensions}. The following path are invalid: """.stripMargin
+      val error =
+        badEntries.map(_.toString() + "\n").mkString(prefix, " - ", "")
+      Configured.notOk(ConfError.message(error))
+    }
+  }
+
+  private val validClasspathFileExtension = List("jar", "zip", "semanticdb")
```

Comment:
why semanticdb? it's a valid file inside an archive or a directory but looking at https://github.com/scalameta/scalameta/blob/c327c3d08b031f560eac9cde3e4906a58b8676fd/semanticdb/metacp/src/main/scala/scala/meta/internal/classpath/ClasspathIndex.scala#L101-L105, I assume scalameta would fail if it encounters it

Reply:
you are absolutely right, I did not try it, I just assumed
good catch